test/support/iterator_utils.h

```diff
@@ -377,39 +378,41 @@ struct iterator_invoker
     ::std::enable_if_t<is_base_of_iterator_category<::std::random_access_iterator_tag, Iterator>::value>
     operator()(Policy&& exec, Op op, Iterator begin, typename ::std::iterator_traits<Iterator>::difference_type n, Rest&&... rest)
     {
-        invoke_if<Iterator>()(n <= sizeLimit, op, exec, make_iterator<Iterator>()(begin), n,
-                              ::std::forward<Rest>(rest)...);
+        invoke_if<Iterator>()(n <= sizeLimit<typename std::iterator_traits<Iterator>::difference_type>, op, exec,
```

Comment:
`sizeLimit<decltype(n)>` is simpler, but it is up to you.

Reply:
Done